ntlm.go

```diff
@@ -212,11 +309,22 @@ func (auth *ntlmAuth) NextBytes(bytes []byte) ([]byte, error) {
 	var lm, nt []byte
 	if (flags & _NEGOTIATE_EXTENDED_SESSIONSECURITY) != 0 {
```

Comment:
Starting here, we see the server wants to neg session security, and we want to populate lm and nt messages.

If we pulled this out into a separate function, we can use a pattern of early returns to remove these nested if/elses, which makes reading harder. Now that you are adding a nested if/else, please try to do so.

Reply:
I refactored the structure of everything around this spot to make it more readable. Hopefully i succeeded! (: